tests/pytests/test_issues.py

```diff
@@ -822,127 +822,60 @@ def test_mod5791(env):
+    conn = env.getConnection()
+    initial_shards_count = env.shardsCount
+    # The first two fields in the cluster info reply are the number of partition in thr cluster.
+    env.assertEqual(conn.execute_command("search.clusterinfo")[:2], ['num_partitions', int(initial_shards_count)])
+
+    # Add a new shard to the cluster. Internally we call CLUSTER MEET to connect the new shard
+    # to the cluster. Also, we internally wait for the cluster to be ready and call "search.CLUSTERREFRESH"
+    # and update the topology change in the new shard (this is where we had a crash in MOD-5778).
+    env.addShardToClusterIfExists()
+    new_shard_conn = env.getConnection(shardId=initial_shards_count+1)
+    # Expect that the cluster will be aware of the new shard, but for redisearch coordinator, the new shard doesn't
+    # consider part of the partition yet as it does not contain any slot.
+    env.assertEqual(int(new_shard_conn.execute_command("cluster info")['cluster_known_nodes']), initial_shards_count+1)
+    env.assertEqual(new_shard_conn.execute_command("search.clusterinfo")[:2], ['num_partitions', int(initial_shards_count)])
+
+    # Move one slot (0) to the new shard (according to https://redis.io/commands/cluster-setslot/)
+    new_shard_id = new_shard_conn.execute_command('CLUSTER MYID')
+    env.assertEqual(new_shard_conn.execute_command(f"CLUSTER SETSLOT 0 NODE {new_shard_id}"), "OK")
+    env.assertEqual(conn.execute_command(f"CLUSTER SETSLOT 0 NODE {new_shard_id}"), "OK")
+
+    # Now we expect that the new shard will be a part of the cluster partition in redisearch (allow some time
+    # for the cluster refresh to occur and acknowledged by all shards)
+    with TimeLimit(40, "topology change was not acknowledged by the new shard"):
+        while True:
+            time.sleep(0.5)
+            cluster_info = new_shard_conn.execute_command("search.clusterinfo")
+            if cluster_info[:2] == ['num_partitions', int(initial_shards_count+1)]:
+                break
+
+    # search.clusterinfo response format is the following:
+    # ['num_partitions', 4, 'cluster_type', 'redis_oss', 'hash_func', 'CRC16', 'num_slots', 16384, 'slots',
+    # [0, 0, ['1f834c5c207bbe8d6dab0c6f050ff06292eb333c', '127.0.0.1', 6385, 'master self']],
+    # [1, 5461, ['60cdcb85a8f73f87ac6cc831ee799b75752aace3', '127.0.0.1', 6379, 'master ']],
+    # [5462, 10923, ['6b2af643a4d6f1723ff2b18b45216d1e0dc7befa', '127.0.0.1', 6381, 'master ']],
+    # [10924, 16383, ['4e51033405651441a4be6ddfb46cd85d0c54af6f', '127.0.0.1', 6383, 'master ']]]
+    unique_shards = set(shard[2][0] for shard in cluster_info[9:])
+    env.assertEqual(len(unique_shards), initial_shards_count+1, message=f"cluster info is {cluster_info}")
+
+    # Verify that slot 0 moved to the new shard,
+    shards_with_slot_0 = [shard for shard in cluster_info[9:] if shard[0] == 0]
+    env.assertEqual(len(shards_with_slot_0), 1, message=f"cluster info is {cluster_info}")
```

Comment:
Is it possible to have more than one shard with slot 0?

Reply:
No and if so it's an error. I did it like this so we will know where is the indexing error if we get one (as a lesson from the previous iterations...) - whether it is in the list comprehension or because we got an empty list